compiler/passes/convert-uast.cpp

```diff
@@ -2325,7 +2383,21 @@ struct Converter {
       }
     }
 
-    Expr* initExpr = convertExprOrNull(node->initExpression());
+    Expr* initExpr = nullptr;
+
+    if (const uast::Expression* ie = node->initExpression()) {
+      if (node->kind() == uast::Variable::TYPE) {
+        if (const uast::BracketLoop* bkt = ie->toBracketLoop()) {
+          initExpr = convertArrayType(bkt);
+        } else {
+          initExpr = convertExprOrNull(node->initExpression());
+        }
+      } else {
+        initExpr = toExpr(convertAST(ie));
+      }
+    } else {
+      initExpr = convertExprOrNull(node->initExpression());
+    }
```

Comment:
This section looks unnecessarily complicated to me. It could just be

```
    if (auto ie = node->initExpression()) {
      auto bkt = ie->toBracketLoop();
      if (bkt && node->kind() == uast::Variable::TYPE) {
        initExpr = convertArrayType(bkt);
      } else {
        initExpr = convertAST(ie);
      }
```